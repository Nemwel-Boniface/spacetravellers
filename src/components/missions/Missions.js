import React from 'react';
import { useSelector, shallowEqual } from 'react-redux';
import MissionSkeleton from './MissionSkeleton';

const Missions = () => {
  const missions = useSelector((state) => state.missionReducer, shallowEqual);

  return (
    <div className="super-mission">
      {missions.map((mission) => (
        <MissionSkeleton
          key={mission.id}
          name={mission.name}
          description={mission.description}
          status={mission.reserved}
        />
      ))}
    </div>
  );
};

export default Missions;
