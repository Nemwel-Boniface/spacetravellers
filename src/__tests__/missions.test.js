import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configStore';
import Missions from '../components/missions/missions';

describe('A tests for my missions component', () => {
  it('Check whether mission component renders', () => {
    const mission = {
      id: 1,
      name: 'Thaicom',
      description: 'Thaicom is the name of a series of  communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
      reserved: false,
    };
    const missionTree = renderer.create(
      <Provider store={store}>
        <Missions
          key={mission.id}
          description={mission.description}
          name={mission.name}
          reserved={mission.reserved}
        />
      </Provider>,
    );

    expect(missionTree).toMatchSnapshot();
  });

  it('We now check whether it renders the mission correctly', () => {
    const missionTree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(missionTree).toMatchSnapshot();
  });
});
