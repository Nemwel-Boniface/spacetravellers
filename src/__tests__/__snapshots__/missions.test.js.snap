// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A tests for my Rockets component Check whether Rockets component renders 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;

exports[`A tests for my Rockets component We now check whether it renders the rockets correctly 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;

exports[`A tests for my missions component Check whether Rockets component renders 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;

exports[`A tests for my missions component Check whether mission component renders 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;

exports[`A tests for my missions component We now check whether it renders the mission correctly 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;

exports[`A tests for my missions component We now check whether it renders the rockets correctly 1`] = `
<div
  className="super-mission"
>
  <div
    className="header-container"
  >
    <p>
      Mission
    </p>
    <p
      className="hola"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      JOin
    </p>
  </div>
  <div
    className="mission-container"
  />
</div>
`;
